### PR TITLE
Refactor deck to use prowjob

### LINKED ISSUE #5216: Use ProwJobs in deck
deck is using its own format for exposing jobs that predates ProwJobs. We should get rid of the old format in favor of using ProwJobs.

TODO:
- [x] Expose ProwJobs in deck at `/prowjobs.js` or something. (https://github.com/kubernetes/test-infra/pull/5631)
- [ ] Switch the JS to look at those. (https://github.com/kubernetes/test-infra/pull/7301)
- [ ] Switch the flake detector stuff to look at those.
- [ ] (fwiw) Switch the [submit queue](https://github.com/kubernetes/test-infra/blob/b666c823028538796261104b1f074b5a76183002/mungegithub/mungers/submit-queue.go#L708) to use the new endpoint and a correct [timestamp format](https://github.com/kubernetes/test-infra/blob/b666c823028538796261104b1f074b5a76183002/mungegithub/mungers/submit-queue.go#L743).
- [ ] Delete `/data.js` and associated machinery.

/area prow

### PR DESCRIPTION
**Pull request by clarketm**
Targets ~2~ 1 sub-issues of #5216 -> 
1. `"Switch the JS to look at prowjobs."` https://github.com/kubernetes/test-infra/issues/5216#issue-269051738, replacing outdated PR #7301.
2. ~`"Delete /data.js and associated machinery."`~

image

### CODE REVIEW

**Comment by clarketm** [comment]
@Katharine - regarding doubling the data. What are your thought on pruning the payload? [example](https://github.com/kubernetes/test-infra/issues/5216#issuecomment-526289836)

File: prow/cmd/deck/static/prow/prow.ts [hunk 1]
```diff
@@ -460,62 +483,72 @@ function redraw(fz: FuzzySearch): void {
...
+        jobCountMap.set(state, (jobCountMap.get(state) || 0) + 1);
 
         // accumulate a count of the percentage of successful jobs over each interval
-        const started = Number(build.started);
+        const started = Date.parse(startTime) / 1000;
+        const finished = Date.parse(completionTime) / 1000;
+        // const finished = completionTime ? Date.parse(completionTime): now;
+
+        const durationSec = completionTime ? finished - started : 0;
+        const durationStr = completionTime ? ms(durationSec * 1000) : "";
```
[the remark this answers is not in the document]
Comment (clarketm) [reply prow/cmd/deck/static/prow/prow.ts]: 🤦‍♂ You are right. I was mistakengly looking at [parseDuration](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/deck/static/prow/prow.ts#L814-L834). I will kill the library and use this instead.

File: prow/cmd/deck/main.go [hunk 2]
```diff
@@ -469,7 +469,6 @@ func prodOnlyMain(cfg config.Getter, o options, mux *http.ServeMux) *http.ServeM
 	cfgGetter := func() *prowapi.RerunAuthConfig { return &cfg().Deck.RerunAuthConfig }
 
 	// setup prod only handlers
-	mux.Handle("/data.js", gziphandler.GzipHandler(handleData(ja)))
```
[the remark this answers is not in the document]
Comment (clarketm) [reply prow/cmd/deck/main.go]: I updated to preserve the `/data.js` endpoint.

File: prow/cmd/deck/static/prow/prow.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (clarketm) [reply prow/cmd/deck/static/prow/prow.ts]: Switched to using `formatDuration`. Thanks again for pointing that out.

**Comment by clarketm** [comment]
@stevekuznetsov @Katharine - PTAL

**Updates**:
1. Preserving `data.js` endpoint for backwards compat.
2. Accept additional `omit` values as query param (e.g. `omit=annotations,labels,decoration_config,pod_spec`) to limit payload size https://github.com/kubernetes/test-infra/issues/5216#issuecomment-526289836.
3. Remove additional external libraries/dependencies.

**Comment by clarketm** [comment]
@Katharine - is there any additional refactoring you want me to do here (e.g. update the variable names in the destructuring)?

I should have addressed all change suggestions: https://github.com/kubernetes/test-infra/pull/14126#issuecomment-526374805

**Comment by alvaroaleman** [comment]
/uncc
This is mostly TS which I can not review.

**Pull request merged**